### ISSUE TITLE
[release/2.12] PT2E INT8 performance regression on resnext50_32x4d (XPU): ~12% Asymm / ~13% Symm slower on 2.12.0+xpu vs release/2.11

### ISSUE THREAD

**Issue by mengfei25** [intel/torch-xpu-ops#3491]
### Summary
On `torch==2.12.0+xpu`, PT2E (inductor) INT8 throughput on XPU regresses ~12–13% on `resnext50_32x4d` versus `release/2.11`, while FP32 throughput is unchanged. Both `XPU_QUANT_CONFIG=ASYMM` and `XPU_QUANT_CONFIG=SYMM` are affected, so this is not config-specific — it points at the shared INT8 (oneDNN graph) lowering / kernel selection on XPU.

### Environment
- `torch`: `2.12.0+xpu` (target) vs `release/2.11` (baseline)
- Device: XPU
- Benchmark harness: `https://github.com/zxd1997066/benchmark` branch `yifeng/pt2e_xpu` cloned as `pt2e-performance`
- Mode: eval, channels-last, `--torchdynamo inductor`, PT2E quantization

### Reproduce
```bash
git clone -b yifeng/pt2e_xpu https://github.com/zxd1997066/benchmark pt2e-performance

PERF_CMD="python pt2e-performance/run_benchmark.py xpu \
          --test eval --channels-last --metrics throughputs --torchdynamo inductor"
m=resnext50_32x4d

# FP32 (unchanged)
${PERF_CMD} -m "${m}"

# INT8 ASYMM (regressed)
XPU_QUANT_CONFIG=ASYMM ${PERF_CMD} -m "${m}" --quantization pt2e

# INT8 SYMM (regressed)
XPU_QUANT_CONFIG=SYMM  ${PERF_CMD} -m "${m}" --quantization pt2e
```
Throughput is parsed from `pt2e-performance/.userbenchmark/**/eval_throughput`.

### Results

`resnext50_32x4d` throughput (eval, channels-last, inductor):

| Mode       |  target / baseline |
|------------|-----------------:|
| FP32       |             1.003 |
| Asymm INT8 |              **0.884** |
| Symm INT8  |            **0.868** |

→ INT8 regression **−11.6% (Asymm)** and **−13.2% (Symm)**; FP32 within noise.

### Analysis (PyTorch + torch-xpu-ops code refs)

The PT2E INT8 path on XPU goes through the shared mkldnn lowerings and dispatches to oneDNN graph extern kernels:

1. **Quantized-conv lowering is shared CPU/XPU** — `torch/_inductor/mkldnn_ir.py` (`QConvPointWisePT2E.create` → `_prepare_convolution_fusion_create`), gated by `SUPPORTED_MKLDNN_DEVICES = ("cpu", "xpu")` in `torch/_inductor/utils.py`. The XPU side resolves to `aoti_torch_xpu__qconv_pointwise_tensor`. Any change to the shared lowering between 2.11 and 2.12 (channels-last propagation, weight reorder, post-op fusion choice) directly affects XPU throughput.

2. **Algorithm selection for extern quantized kernels** — `torch/_inductor/select_algorithm.py` around L3254 catches `TensorMeta.from_irnodes(...)` failures for the extern choice. If meta is missing on XPU, the autotuner cannot benchmark variants and may fall back to a non-optimal kernel — which would manifest as a perf regression even when accuracy survives.

3. **Weight zero-point materialization** — `torch/_inductor/mkldnn_lowerings.py` near L763 wraps `w_zp` via `torch.tensor(w_zp_tensor, dtype=torch.int32)`. The same regression footprint appearing under both ASYMM and SYMM suggests the perf hit is not in the asymmetric-only zero-point branch but in the common reorder/fusion path.

4. **`resnext50_32x4d` is grouped-conv heavy (cardinality=32)**. Grouped INT8 convolution on XPU is the most layout/format-sensitive path in oneDNN graph; a change in channels-last propagation or grouped-conv fusion eligibility would land here first. The fact that FP32 is unchanged and INT8 is regressed for both quant configs strongly localizes the regression to the INT8 conv lowering / fusion / kernel selection — not to dataloader, dynamo tracing, or post-grad passes.

### Likely root-cause hypotheses (ranked)
1. **Grouped INT8 conv fusion / channels-last propagation regression** in the shared `_prepare_convolution_fusion_create` path between 2.11 → 2.12 (selecting a slower oneDNN primitive or losing post-op fusion on grouped conv).
2. **Extern-choice autotuner degradation on XPU** due to `TensorMeta.from_irnodes` failures (`select_algorithm.py` L3254) leading to suboptimal kernel selection for `qconv_pointwise`.
3. **Weight reorder regression** for INT8 grouped weights on XPU (the `w_zp` rewrap path in `mkldnn_lowerings.py` near L763 indicates the constant-materialization branch is exercised every run).

### Related
- Accuracy regression (likely same lowering area): intel/torch-xpu-ops#3489

**Comment by chuanqi129** [comment]
@kaileiyx let's check the latest nightly/weekly

**Comment by kaileiyx** [comment]
PT2.13 result (PT2.13 vs. PT2.12)
<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/kaileiyx/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/kaileiyx/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">



</head>

<body link="#0563C1" vlink="#954F72">


Mode | target /   baseline
-- | --
FP32 | 1.000
Asymm INT8 | 0.977
Symm INT8 | 0.979



</body>

</html>

Latest Nightly result (Latest Nightly vs. PT2.13)
<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/kaileiyx/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/kaileiyx/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">



</head>

<body link="#0563C1" vlink="#954F72">


Mode | target /   baseline
-- | --
FP32 | 1.000
Asymm INT8 | 1.012
Symm INT8 | 1.023



</body>

</html>

**Comment by chuanqi129** [comment]
Close it due to unreproduced now
